### DISCUSSION TITLE
Conduct Physical Simulation (Cloth simulation) crashes

### DISCUSSION THREAD

**Discussion by AlkisPis** [taichi-dev/taichi#8593] [Q&A]
Re: Conduct Physical Simulation in https://docs.taichi-lang.org/docs/cloth_simulation

I use `ti.init(arch=ti.cpu) ` for initialization --as the document suggests-- since Vulkan is not installed in my system.

After some point in the simulation the script crashes, issuing the following error:
**"RHI Error: (-4) Vulkan device might be lost (vkQueueSubmit failed)
Assertion failed: false && "Error without return code", file C:\Users\buildbot\actions-runner\_work\taichi\taichi\taichi\rhi\vulkan\vulkan_device.cpp, line 2038"**

Does this happen to everybody or only in my place?

**Comment by AlkisPis** [comment]
I replaced `scene = ti.ui.Scene()` with `scene = window.get_scene()`. It works now.

**Reply by AlkisPis** [reply]
Well, I did a few more tests and realized that this script is HIGHLY UNSTABLE.  Sometimes it is run without any problem and other times it crashes a couple of seconds after the animation starts! And it crashes real hard! The computer stops responding for quite a while. Then the script terminates by itself, issuing the message I included in my post. This is a real nightmare for a programmer! 

Otherwise, it's a great taichi Example and it would be great is someone  shows up with a solution or at least  a reason of such an unstable behavior ...

**Comment by Vlada3333** [comment]
It is necessary to finalize.
